### PR TITLE
Adapt test_report.py to support dask/dask repository

### PR DESCRIPTION
**Pull request by milesgranger**
Part of https://github.com/dask/dask/issues/10732

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

Adapts the `continuous_integration/scripts/test_report.py` to support test reporting added by https://github.com/dask/dask/pull/10769. Idea being to add a step later in dask/dask to fetch this script and run it over there.

Example for --repo dask/distributed --branch main (no changes)
distributed-report

Example for --repo milesgranger/dask --branch failure-test
dask-report

### CODE REVIEW

File: continuous_integration/scripts/test_report.py [hunk 1]
```diff
@@ -337,9 +369,11 @@ def download_and_parse_artifacts(
         print(f"Downloading and parsing {nartifacts} artifacts...")
 
         for r in runs:
-            jobs_df = get_jobs(r, session=session)
+            jobs_df = get_jobs(r, session=session, repo=repo)
             r["dfs"] = []
             for a in r["artifacts"]:
+                if repo.endswith("/dask") and not a["name"].startswith("test-results-"):
+                    continue
```
Comment (crusaderky) [inline continuous_integration/scripts/test_report.py:376]:
```suggestion
```
Adjust to https://github.com/dask/dask/pull/10781

File: continuous_integration/scripts/test_report.py [hunk 2]
```diff
@@ -140,17 +141,48 @@ def get_jobs(run, session):
...
-    # See `Set $TEST_ID` step in `tests.yaml`
-    name_components["partition"] = name_components.partition.str.replace(" ", "")
+        # See `Set $TEST_ID` step in `tests.yaml`
+        name_components.columns = ["OS version", "environment", "option"]
+        df_jobs["suite_name"] = name_components.iloc[:, 0].str.cat(
+            name_components.iloc[:, 1:], sep="-"
+        )
+        df_jobs["suite_name"] = ("test-results-" + df_jobs.suite_name).str.replace(
+            "--", "-"
+        )
```
Comment (crusaderky) [inline continuous_integration/scripts/test_report.py:167]:
```suggestion
        df_jobs["suite_name"] = df_jobs.suite_name.str.replace("--", "-")
```
Adjust to https://github.com/dask/dask/pull/10781

**Comment by milesgranger** [comment]
@crusaderky I ran it once more after your updates in dask and seems to work still:

https://github.com/milesgranger/dask/pull/2

image

**Comment by fjetter** [comment]
> Idea being to add a step later in dask/dask to fetch this script and run it over there.

I suggest to just copy the script.

**Comment by fjetter** [comment]
is this good to go?

**Comment by milesgranger** [comment]
I'm happy with it.

**Comment by fjetter** [comment]
> > Idea being to add a step later in dask/dask to fetch this script and run it over there.
>
> > I suggest to just copy the script.

I don't have a strong opinion. https://github.com/dask/dask/pull/10783 looks straight forward but I would like to avoid us having to invest a lot of time maintaing compatibility. if it works now, we can keep it as is

**Comment by fjetter** [comment]
I might've merged a little prematurely. When I run the script I get

```
 python continuous_integration/scripts/test_report.py --max-days 20 --max-runs 5 --nfails 1 -o test_report.html --repo=dask/dask

Getting list of workflow runs...
Found 26 workflow runs
Fetching artifact listing for the 5 most recent workflow runs
Downloading and parsing 80 artifacts...
Traceback (most recent call last):
  File "/Users/fjetter/workspace/distributed/continuous_integration/scripts/test_report.py", line 536, in <module>
    main()
  File "/Users/fjetter/workspace/distributed/continuous_integration/scripts/test_report.py", line 460, in main
    dfs = list(
  File "/Users/fjetter/workspace/distributed/continuous_integration/scripts/test_report.py", line 372, in download_and_parse_artifacts
    jobs_df = get_jobs(r, session=session, repo=repo)
  File "/Users/fjetter/workspace/distributed/continuous_integration/scripts/test_report.py", line 148, in get_jobs
    df_jobs.name.str.replace("(", ",")
  File "/Users/fjetter/miniforge3/envs/dask-distributed/lib/python3.10/site-packages/pandas/core/frame.py", line 10034, in apply
    return op.apply().__finalize__(self, method="apply")
  File "/Users/fjetter/miniforge3/envs/dask-distributed/lib/python3.10/site-packages/pandas/core/apply.py", line 837, in apply
    return self.apply_standard()
  File "/Users/fjetter/miniforge3/envs/dask-distributed/lib/python3.10/site-packages/pandas/core/apply.py", line 965, in apply_standard
    results, res_index = self.apply_series_generator()
  File "/Users/fjetter/miniforge3/envs/dask-distributed/lib/python3.10/site-packages/pandas/core/apply.py", line 981, in apply_series_generator
    results[i] = self.func(v, *self.args, **self.kwargs)
  File "/Users/fjetter/workspace/distributed/continuous_integration/scripts/test_report.py", line 153, in <lambda>
    (row[1].strip(), row[2].strip(), row[3].strip() or "-")
AttributeError: 'NoneType' object has no attribute 'strip'
```

**Comment by milesgranger** [comment]
I'll fix in a follow up.

**Pull request merged**